File: BEmuJava/bemu/src/main/java/com/bloomberglp/blpapi/MarketElementNull.java
//------------------------------------------------------------------------------
// <copyright project="BEmu_maven" file="/BEmu_maven/bemu/src/main/java/com/bloomberglp/blpapi/MarketElementNull.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bloomberglp.blpapi;

import com.bloomberglp.blpapi.Name;

public class MarketElementNull extends Element
{
	private final String _name;
	
	MarketElementNull(String name)
	{
		this._name = name;
	}
	
    public Name name() throws Exception
    {
    	return new Name(this._name);
    }
    
    public int numValues()
    {
    	return 0;
    }
    
    public int numElements()
    {
    	return 0;
    }
    
    public boolean isComplexType()
    {
    	return false;
    }
    
    public boolean isArray()
    {
    	return true;
    }
    
    public boolean isNull()
    {
    	return true;
    }
    
    protected StringBuilder prettyPrint(int tabIndent)
    {
    	return new StringBuilder();
    }
}

File: BEmu_cpp/src/MarketDataRequest/MarketElementNull.cpp
//------------------------------------------------------------------------------
// <copyright project="BEmu_cpp" file="src/MarketDataRequest/MarketElementNull.cpp" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

#include "MarketDataRequest/MarketElementNull.h"

namespace BEmu
{
	namespace MarketDataRequest
	{
		MarketElementNull::MarketElementNull(const std::string& name)
		{
			this->_name = name;
		}

		Name MarketElementNull::name() const
		{
			return Name(this->_name.c_str());
		}

		size_t MarketElementNull::numValues() const
		{
			return 0;
		}

		size_t MarketElementNull::numElements() const
		{
			return 0;
		}

		bool MarketElementNull::isNull() const
		{
			return true;
		}

		bool MarketElementNull::isArray() const
		{
			return true;
		}

		bool MarketElementNull::isComplexType() const
		{
			return false;
		}

		std::ostream& MarketElementNull::print(std::ostream& stream, int level, int spacesPerLevel) const
		{
			return stream;
		}

	}
}